router/utils/utils.go

```diff
@@ -47,9 +46,40 @@ func getRetentionTimeForDestination(destID string) time.Duration {
 	return config.GetDurationVar(720, time.Hour, "Router."+destID+".jobRetention", "Router.jobRetention")
 }
 
-func ToBeDrained(job *jobsdb.JobT, destID, toAbortDestinationIDs string, destinationsMap map[string]*DestinationWithSources) (bool, string) {
+type JobParameters struct {
+	SourceID                string      `json:"source_id"`
+	DestinationID           string      `json:"destination_id"`
+	ReceivedAt              string      `json:"received_at"`
+	TransformAt             string      `json:"transform_at"`
+	SourceTaskRunID         string      `json:"source_task_run_id"`
+	SourceJobID             string      `json:"source_job_id"`
+	SourceJobRunID          string      `json:"source_job_run_id"`
+	SourceDefinitionID      string      `json:"source_definition_id"`
+	DestinationDefinitionID string      `json:"destination_definition_id"`
+	SourceCategory          string      `json:"source_category"`
+	RecordID                interface{} `json:"record_id"`
+	MessageID               string      `json:"message_id"`
+	EventName               string      `json:"event_name"`
+	EventType               string      `json:"event_type"`
+	WorkspaceID             string      `json:"workspaceId"`
+	RudderAccountID         string      `json:"rudderAccountId"`
+}
+
+type AbortConfigs struct {
```

Comment:
We could get rid of this config struct if we had a `Drainer` interface with a single method

```go
Drain(job *jobsdb.JobT, jobParams JobParameters, destinationsMap map[string]*DestinationWithSources) (bool, string)
```
This component could be initialised using the config object. Drain logic and configuration could then be internal to this component

Reply:
Please take another look(rt, brt, router/utils).
Processor changes will follow soon.